monitor-function-calls/src/agent.spec.js

```diff
@@ -0,0 +1,621 @@
+  };
+}
+
+function createMockArgs(functionObject) {
+  const argValues = [];
+  const argNames = [];
+  const argTypes = [];
+
+  functionObject.inputs.forEach((entry) => {
+    argNames.push(entry.name);
+    argTypes.push(entry.type);
+    switch (entry.type) {
+      case 'uint256':
+        argValues.push(0);
+        break;
+      case 'uint256[]':
+        argValues.push([0]);
+        break;
+      case 'address':
+        argValues.push(ethers.constants.AddressZero);
+        break;
+      case 'address[]':
+        argValues.push([ethers.constants.AddressZero]);
+        break;
+      case 'bytes':
+        argValues.push('0xff');
+        break;
+      case 'bytes[]':
+        argValues.push(['0xff']);
+        break;
+      case 'bytes32':
+        argValues.push(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF);
+        break;
+      case 'bytes32[]':
+        argValues.push([0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF]);
+        break;
+      case 'string':
+        argValues.push('placeholderstring');
+        break;
+      case 'string[]':
+        argValues.push(['placeholderstring']);
+        break;
+      case 'tuple':
+        throw new Error('tuple not supported yet');
+      default:
+        throw new Error(`Type passed in is not support: ${entry.type}`);
+    }
+  });
+
+  return { argValues, argNames, argTypes };
+}
+
+function overrideArgument(argValues, argNames, argOverride) {
+  // if an argument override is specified, attempt to set it
+  if (argOverride !== undefined) {
+    if (argOverride.name === undefined) {
+      throw new Error('Must specify name field in argOverride Object');
+    }
+
+    // determine if the argument name specified is in the Array of argument names
+    const argIndex = argNames.indexOf(argOverride.name);
+    if (argIndex === -1) {
+      throw new Error(`Argument name specified in argOverride variable not found in function ABI. argOverride name: ${argOverride.name}, argNames: ${argNames}`);
+    }
+
+    // check that a value is specified in the argOverride Object
+    if (argOverride.value === undefined) {
+      throw new Error('Must specify a value field in argOverride Object');
+    }
+
+    // update the argument value
+    // eslint-disable-next-line no-param-reassign
+    argValues[argIndex] = argOverride.value;
+  }
+}
+
+// set up test configuration parameters that won't change with each test
+// grab the first entry from the 'contracts' key in the configuration file
+const { contracts: configContracts } = config;
+
+let testConfig = {};
+
+const contractName = Object.keys(configContracts)[0];
+const { abiFile, functions } = configContracts[contractName];
+const validContractAddress = configContracts[contractName].address;
+
+const abi = utils.getAbi(abiFile);
+const functionObjects = getObjectsFromAbi(abi, 'function');
+
+// create a fake function name
+function getRandomCharacterString(numCharacters) {
```

Comment:
Moving this helper function and other functions such as `overrideArgument`, `createMockArgs`, etc. into a separate file would make this test file cleaner